kafka-connect-neo4j/src/main/kotlin/streams/kafka/connect/sink/Neo4jService.kt

```diff
@@ -116,29 +116,34 @@ class Neo4jService(private val config: Neo4jSinkConnectorConfig):
     suspend fun writeData(data: Map<String, List<List<StreamsSinkEntity>>>) = coroutineScope {
         val timeout = config.batchTimeout
         val ticker = ticker(timeout)
-        val deferredList = data
-                .flatMap { (topic, records) ->
-                    records.map { async(Dispatchers.IO) { writeForTopic(topic, it) } }
+
+        if (config.parallelBatches) {
+            val deferredList = data
+                    .flatMap { (topic, records) ->
+                        records.map { async(Dispatchers.IO) { writeForTopic(topic, it) } }
+                    }
+            whileSelect {
+                ticker.onReceive {
+                    if (log.isDebugEnabled) {
+                        log.debug("Timeout $timeout occurred while executing queries")
+                    }
+                    deferredList.forEach { deferred -> deferred.cancel() }
```

Comment:
I think changing the order and pulling cancel upfront before the completed check will cancel everything all the time. 
We should put this back behind the completed check.

I'e just move the async/non-async code into a conditional for the two cases.

We might have another option which is changing the way how this is executed with the pool.